Registry/RegistryManager/src/test/java/eu/neclab/ngsildbroker/registryhandler/service/CSourceServiceTest.java

```diff
@@ -1,181 +1,401 @@
+			result.put(AppConstants.INTERNAL_NULL_KEY, entities);
+			when(csourceInfoDAO.getAllEntities()).thenReturn(result);
+			ReflectionTestUtils.setField(csourceService, "AUTO_REG_STATUS", "active");
+			Method postConstruct = CSourceService.class.getDeclaredMethod("loadStoredEntitiesDetails"); // methodName,parameters
+			postConstruct.setAccessible(true);
+			postConstruct.invoke(csourceService);
+			multimaparr.put("content-type", "application/json");
+			UpdateResult updateResult = new UpdateResult();
+			Gson gson = new Gson();
+			Map<String, Object> resolved = gson.fromJson(updatePayload, Map.class);
+			Map<String, Object> originalresolved = gson.fromJson(payload, Map.class);
+			String[] optionsArray = new String[0];
+			AppendCSourceRequest request = new AppendCSourceRequest(multimaparr,
+					"urn:ngsi-ld:ContextSourceRegistration:csr3", originalresolved, resolved, optionsArray);
+			when(csourceInfoDAO.getEntity(any(), any())).thenReturn(payload);
+			updateResult = csourceService.appendToEntry(multimaparr, "urn:ngsi-ld:ContextSourceRegistration:csr3",
+					resolved, optionsArray);
+			Assert.assertEquals(updateResult.getUpdated(), request.getUpdateResult().getUpdated());
+
+		} catch (Exception ex) {
+			Assert.fail();
+		}
+	}
+
+	/**
+	 * this method is use for update CSource Registry if ID is not exist
+	 */
+	@Test
+	public void updateCSourceRegIdNotExistTest() {
+		try {
+
+			ArrayListMultimap<String, String> csourceIds = ArrayListMultimap.create();
+			csourceIds.put(AppConstants.INTERNAL_NULL_KEY, "urn:ngsi-ld:ContextSourceRegistration:csr4");
+			// when(csourceInfoDAO.getAllIds()).thenReturn(csourceIds);
+			HashMap<String, List<String>> result = new HashMap<String, List<String>>();
+			List<String> entities = new ArrayList<String>();
+			result.put(AppConstants.INTERNAL_NULL_KEY, entities);
+			when(csourceInfoDAO.getAllEntities()).thenReturn(result);
+			ReflectionTestUtils.setField(csourceService, "AUTO_REG_STATUS", "active");
+			Method postConstruct = CSourceService.class.getDeclaredMethod("loadStoredEntitiesDetails"); // methodName,parameters
+			postConstruct.setAccessible(true);
+			postConstruct.invoke(csourceService);
+			multimaparr.put("content-type", "application/json");
+			UpdateResult updateResult = new UpdateResult();
+			Gson gson = new Gson();
+			Map<String, Object> resolved = gson.fromJson(updatePayload, Map.class);
+			Map<String, Object> originalresolved = gson.fromJson(payload, Map.class);
+			String[] optionsArray = new String[0];
+			AppendCSourceRequest request = new AppendCSourceRequest(multimaparr,
+					"urn:ngsi-ld:ContextSourceRegistration:csr3", originalresolved, resolved, optionsArray);
+			when(csourceInfoDAO.getEntity(any(), any())).thenReturn(payload);
+			updateResult = csourceService.appendToEntry(multimaparr, "urn:ngsi-ld:ContextSourceRegistration:csr3",
+					resolved, optionsArray);
+			Assert.assertEquals(updateResult.getUpdated(), request.getUpdateResult().getUpdated());
```

Comment:
[Major] as per the method name indication, This test case for check the id exist on not but you have compared object but it should be compare like 404 or resource not found.

Reply:
in this case update result return array list object so for this we compare array list result